### ISSUE TITLE
`clang-format` treats `in(arg)` as a contextual keyword, misclassifies following binary `-` as unary (no space)

### ISSUE THREAD

**Issue by alex-tee** [llvm/llvm-project#206339]
> **⚠️ Disclosure: This bug report was AI-generated based on my input.**
>
> The reproduction commands, the bisection across identifier substitutions, the version/platform info, and the prose below were all produced by an AI assistant (GLM-5.2 via opencode) in response to my request to investigate why `-` is formatted incorrectly in some cases when using clang-format. The reproductions were run on my machine locally to confirm the bug and when it occurs, but I have not verified any of the hypotheses in this report.
>
> Please treat the "Hypothesis" section as speculation pointing at a likely region of the code, not as a confirmed root cause — I have not read the relevant clang-format source to verify the mechanism described.

---

## Summary

When the lowercase identifier `in` is immediately followed by a non-empty parenthesized argument list — e.g. `in(x)`, `obj.in(x)`, `ns::in(x)` — clang-format's `TokenAnnotator` appears to treat `in` as a contextual keyword (likely inherited from Objective-C / Pascal-style `for (... in ...)` fast enumeration). As a side effect, any binary `-` operator immediately following the call is misclassified as **unary** minus, and the required space before it is dropped.

This causes incorrect formatting anywhere the `au` units library's `.in(unit)` conversion API is followed by subtraction — a very common pattern in real C++ code.

## Platform / Version

```
clang-format version 22.1.6
```
Arch Linux package, x86_64.

### Minimal `.clang-format`

Empty style (i.e. `clang-format -style="{}"`) is sufficient to reproduce. The bug also reproduces with LLVM style and with GNU style. It is **independent of every formatting option** I tried (`BreakBeforeBinaryOperators`, `SpaceBeforeParens`, `AlignAfterOpenBracket`, `BinPackArguments`, etc.).

### Input

```cpp
void f() {
  int a = std::max(x.in(y) - w, 1);
}
```

### Actual output

```cpp
void f() {
  int a = std::max(x.in(y) -w, 1);
}
```

Note `-w` — the binary subtraction operator has no space before it.

### Expected output

```cpp
void f() {
  int a = std::max(x.in(y) - w, 1);
}
```

## Investigation: the trigger is specifically `in`

I bisected by substituting the identifier. Only the lowercase identifier `in` (exactly), followed by a non-empty `(...)`, triggers the bug:

```cpp
void f() {
  int a = std::max(x.in(y)   - w, 1); // ❌ -> x.in(y) -w
  int b = std::max(x.out(y)  - w, 1); // ✅ -> x.out(y) - w
  int c = std::max(x.of(y)   - w, 1); // ✅
  int d = std::max(x.is(y)   - w, 1); // ✅
  int e = std::max(x.in2(y)  - w, 1); // ✅
  int g = std::max(x.IN(y)   - w, 1); // ✅ (uppercase)
  int h = std::max(x::in(y)  - w, 1); // ❌ -> x::in(y) -w
  int i = std::max(in(y)     - w, 1); // ❌ -> in(y) -w
}
```

The bug also reproduces regardless of:
- the depth of the member-access chain (`x.in(y)`, `a.b.c.in(y)`, …),
- whether the argument is qualified (`in(units::ticks)`),
- whether the call site is nested inside another call (`std::max(... - w, ...)`),
- whether `in` is reached via `.`, `->`, or `::`.

### Cases that do **not** trigger

- Empty argument list: `x.in() - w` → `x.in() - w` ✅ (correct)
- `in` not followed by parens at all: `x.in - w`, `in - w`, `foo + in - w` → all correct ✅
- Unary minus that *should* have no space (`-w` after a literal): unaffected, still correct.
- Other binary operators after `in(y)`: `x.in(y) + w`, `x.in(y) * w`, `x.in(y) / w` — all formatted correctly with spaces. Only `-` is misclassified.

This last point is what strongly suggests the binary/unary disambiguation for `-` specifically is the culprit, not a general tokenization failure of the `in(...)` expression.

## Hypothesis

In `lib/Format/TokenAnnotator.cpp`, the `in` keyword is recognized in some code paths (e.g. Objective-C fast enumeration `for (x in collection)`). When the annotator sees `in` immediately followed by `(`, it appears to leave the `in` token in a state that causes the following `-` token's `TokenTypes` / previous-token check in `spaceRequiredBetween` (or in the binary-operator detection in `calculateFormattingInformation` / `determineTokenType`) to conclude the `-` is unary.

A likely place to look: the function that decides whether a `-` is `TT_BinaryOperator` vs. `TT_UnaryOperator`, where the previous token is `r_paren` whose matching `l_paren` is preceded by `in`. The `in`-as-keyword flag may be poisoning the lookup of whether the parenthesized expression is a "complete" expression.

## Workaround

End users currently have to do one of:

1. Wrap the call in redundant parentheses: `(x.in(y)) - w` formats correctly.
2. Extract `x.in(y)` into an intermediate variable so the `-` is not adjacent to the call on the same logical line.
3. Avoid the `in` identifier entirely as a member function name — which is not an option for users of the [`au`](https://github.com/aurora-opensource/au) units library, whose public conversion API is precisely `quantity.in(unit)`.

## Impact

This silently produces malformed-looking code anywhere `au`'s `.in(unit)` (or any user-defined `in(...)` member or free function) is followed by arithmetic. The fix would benefit anyone using `au`, which is a widely-adopted strong-units library in the C++ community.
